src/ReflectionParameter.php

```diff
@@ -345,7 +354,7 @@ public function isDefaultValueConstant()
      */
     public function isOptional()
     {
-        return $this->isVariadic() || ($this->isDefaultValueAvailable() && $this->haveSiblingsDefalutValues());
+        return $this->isVariadic() || ($this->isDefaultValueSet() && $this->haveSiblingsDefalutValues());
```

Comment:
Does this belong to this PR or should a separate bugfix PR with connected unit tests?

Reply:
Again, this is part of the handling for optional parameters without default values, which is new in this PR.

Reply:
Better to move this into separate PR for optional parameter without default value handling (part 1).

Reply:
OK. See above.